### PR TITLE
libssh: fix static libs

### PR DESCRIPTION
**Pull request by jeroen**
- [x] Have you followed the [guidelines for contributing](https://github.com/Homebrew/homebrew-core/blob/HEAD/CONTRIBUTING.md)?
- [x] Have you checked that there aren't other open [pull requests](https://github.com/Homebrew/homebrew-core/pulls) for the same formula update/change?
- [x] Have you built your formula locally with `brew install --build-from-source <formula>`, where `<formula>` is the name of the formula you're submitting?
- [x] Is your test running fine `brew test <formula>`, where `<formula>` is the name of the formula you're submitting?
- [x] Does your build pass `brew audit --strict <formula>` (after doing `brew install <formula>`)?

-----

I noticed after upgrading that libssh no longer contains a static library. It turns out the `WITH_STATIC_LIB` parameter no longer works. This fixes the bottle again to contain both the shared and static lib.

It was changed in upstream here: https://github.com/libssh/libssh-mirror/commit/729c92606c88b60e7c255d393756d1d3b7e30698

### CODE REVIEW

File: Formula/libssh.rb [hunk 1]
```diff
@@ -22,10 +23,12 @@ class Libssh < Formula
...
     mkdir "build" do
-      system "cmake", "..", "-DWITH_STATIC_LIB=ON",
-                            "-DWITH_SYMBOL_VERSIONING=OFF",
-                            *std_cmake_args
+      system "cmake", "..", *std_cmake_args, "-DWITH_SYMBOL_VERSIONING=OFF"
       system "make", "install"
+      system "make", "clean"
+      system "cmake", "..", *std_cmake_args, "-DBUILD_SHARED_LIBS=OFF"
+      system "make"
+      lib.install "src/libssh.a"
```
Comment (carlocab) [inline Formula/libssh.rb:31]: Is there no way to build this with only one `cmake` call?

Comment (jeroen) [reply Formula/libssh.rb:30]: Right, let me try again.

Comment (jeroen) [reply Formula/libssh.rb:30]: OK I updated it, thanks!

Comment (carlocab) [reply Formula/libssh.rb:30]: Thanks. Is the `lib.install` line still needed? Shouldn't that be installed by `make install`?

Comment (jeroen) [reply Formula/libssh.rb:30]: Yeah, strangely it doesn't get installed automatically, unless you build with `BUILD_SHARED_LIBS=OFF`.

Comment (carlocab) [reply Formula/libssh.rb:30]: That's a bit troubling. Is it possible upstream actually don't intend for them to be built simultaneously for some reason?

Comment (jeroen) [reply Formula/libssh.rb:30]: It's actually how it works for most libraries, that you can install either shared or static libs, depending on the value of `BUILD_SHARED_LIBS`. They have done this on purpose: https://github.com/libssh/libssh-mirror/commit/729c92606c88b60e7c255d393756d1d3b7e30698

Comment (carlocab) [reply Formula/libssh.rb:30]: Yes, but I was assuming you used to be able to install both... they can't have done this on purpose for no reason, right?

**Comment by jeroen** [comment]
@cryptomilk question about libssh: is it expected that previously `-DWITH_STATIC_LIB=ON` would install a static lib, but with `-DBUILD_STATIC_LIB=ON` the static lib is built but not installed by default? Does the change above look correct to you if we want to install both the shared and static lib?

**Comment by carlocab** [comment]
Marking this as do not merge in the meantime until we understand why their build scripts make it difficult for you to install both the shared and static libraries simultaneously. If there's a reason for that, we need to know what it is before we can proceed.

**Comment by jeroen** [comment]
I don't quite understand, this is the default behavior for almost any cmake library. That's why we usually do two builds, one for the shared libs and one for the static libs (but you requested to do a single build). 

For example, from poppler:

https://github.com/Homebrew/homebrew-core/blob/9adfd41f52709652a623b5b6f99005144fd8e34a/Formula/poppler.rb#L64-L71

Or from opencv:

https://github.com/Homebrew/homebrew-core/blob/9adfd41f52709652a623b5b6f99005144fd8e34a/Formula/opencv.rb#L98-L108

Or jasper:

https://github.com/Homebrew/homebrew-core/blob/581cc33a4422ac232028a0e5f18e9f31cc01597b/Formula/jasper.rb#L24-L37

And many more. There is nothing unusual about this. Anyway I suppose we can wait for upstream to confirm that this is intended.

**Comment by carlocab** [comment]
I guess the difference is that I know that those aren't broken because they've been that way for a while now. I still don't know if this might break something. That said, I'll take off the do not merge flag to avoid deterring someone who actually knows this better than I do from taking a look.

**Comment by jeroen** [comment]
For what it's worth, I just tested my R bindings to work with the static library generated from this PR and it works 💯

**Review by chenrui333 — APPROVED** [review]
lgtm

**Comment by chenrui333** [comment]
> I don't quite understand, this is the default behavior for almost any cmake library. That's why we usually do two builds, one for the shared libs and one for the static libs (but you requested to keep a single build).
> 
> For example, from poppler:
> 
> https://github.com/Homebrew/homebrew-core/blob/9adfd41f52709652a623b5b6f99005144fd8e34a/Formula/poppler.rb#L64-L71
> 
> Or from opencv:
> 
> https://github.com/Homebrew/homebrew-core/blob/9adfd41f52709652a623b5b6f99005144fd8e34a/Formula/opencv.rb#L98-L108
> 
> Or jasper:
> 
> https://github.com/Homebrew/homebrew-core/blob/581cc33a4422ac232028a0e5f18e9f31cc01597b/Formula/jasper.rb#L24-L37
> 
> And many more. There is nothing unusual about this. Anyway I suppose we can wait for upstream to confirm that this default behavior is intended.

that is a good point, we should enable the STATIC_LIB flag for cmake to consolidate the build options for libs

**Comment by cryptomilk** [comment]
You can either build a shared library or a static library. This is controlled by `cmake -DBUILD_SHARED_LIBS=OFF`.

https://cmake.org/cmake/help/latest/variable/BUILD_SHARED_LIBS.html

**Comment by cryptomilk** [comment]
We are following the cmake way which is either shared (default) or static.

**Comment by carlocab** [comment]
Yes, that's essentially what you already said in your first comment. Would you know if building both in separate builds but installing them to the same prefix can cause any problems?

**Comment by cryptomilk** [comment]
I wouldn't provide a static lib at all. libssh is LGPL and probably people don't understand what static linking against a LGPL library implies ...

**Comment by jeroen** [comment]
Thanks all! Yes we do know about the LGPL, we link this from R which is GPL itself.
